acvm/src/pwg/blackbox/ecdsa.rs

```diff
@@ -27,6 +27,17 @@ pub(crate) fn secp256k1_prehashed(
     hashed_message_inputs: &[FunctionInput],
     output: Witness,
 ) -> Result<OpcodeResolution, OpcodeResolutionError> {
+    let hashed_message: [u8; 32] =
+        to_u8_vec(initial_witness, hashed_message_inputs)?.try_into().map_err(|_| {
```

Comment:
The hashed_message does not need to be 32 bytes. Barretenberg will crash if one supplies a hashed_message thats more than 32 bytes, but the algorithm itself is for a generic hash.

Reply:
Gotcha, I've relaxed this to accept a slice for the message hash then.